src/Koffee/MainView.fs

```diff
@@ -241,21 +241,56 @@ module MainView =
                                 |> Seq.ifEmpty [(' ', "No bookmarks set")]
                             window.Bookmarks.ItemsSource <- bookmarks
                             window.BookmarkPanel.Visible <- true
+                            window.HistoryPanel.Visible <- false
+                        | Prompt ShowHistory ->
+                            let formatIndex char i =
+                                sprintf "%i%c" (i + 1) char
+
+                            let formatPath (p: Path) =
+                                p.Format pathFormat
+
+                            let formatStack char stack =
+                                stack
+                                |> Seq.mapi (fun i (path, _) -> (formatIndex char i, formatPath path))
+
+                            let isEmpty = forwardStack = [] && backStack = []
+                            let collapsedIfEmpty = if isEmpty then Visibility.Collapsed else Visibility.Visible
+                            let visibleIfEmpty = if isEmpty then Visibility.Visible else Visibility.Collapsed
+
+                            window.HistoryError.Visibility <- visibleIfEmpty
+
+                            window.HistoryForward.Visibility <- collapsedIfEmpty
+                            window.HistoryBack.Visibility <- collapsedIfEmpty
+                            window.HistoryCurrent.Visibility <- collapsedIfEmpty
+
+                            if isEmpty then
+                                window.HistoryError.Text <- "Nothing in history"
+                            else
+                                window.HistoryForward.ItemsSource <- Seq.truncate 4 forwardStack |> formatStack 'L'
+                                window.HistoryBack.ItemsSource <- Seq.truncate 9 backStack |> formatStack 'H' |> Seq.rev
```

Comment:
Maybe I should fetch the keycodes from the configured keybindings?
Though I had troubles finding how keybindings are meant to be stored. Any tips? @mattstermiller

Reply:
`KeyBinding.defaultsAsString |> List.find (snd >> (=) Back) |> fst`
- should work fine for now. User-configurable settings are planned later.